File: src/main/java/com/freya02/botcommands/internal/application/slash/autocomplete/ChoiceSupplier.java
package com.freya02.botcommands.internal.application.slash.autocomplete;

import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.Command;

import java.util.Collection;
import java.util.List;

public interface ChoiceSupplier {
	List<Command.Choice> apply(CommandAutoCompleteInteractionEvent event, Collection<?> collection) throws Exception;
}

File: src/main/kotlin/com/freya02/botcommands/internal/commands/application/slash/autocomplete/suppliers/ChoiceSupplier.kt
package com.freya02.botcommands.internal.commands.application.slash.autocomplete.suppliers

import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent
import net.dv8tion.jda.api.interactions.commands.Command

internal sealed interface ChoiceSupplier {
    fun apply(event: CommandAutoCompleteInteractionEvent, collection: Collection<Any>): List<Command.Choice>
}